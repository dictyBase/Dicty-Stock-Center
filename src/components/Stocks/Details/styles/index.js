import { makeStyles } from "@material-ui/core/styles"
import { green } from "@material-ui/core/colors"

const useStyles = makeStyles(({ palette, spacing }) => ({
  root: {
    width: "90%",
    margin: "auto",
  },
  header: {
    textAlign: "center",
  },
  backButton: {
    marginTop: "10px",
    textAlign: "left",
    color: "#5f6368",
    "&:hover": {
      backgroundColor: "transparent",
      color: "inherit",
    },
  },
  cardHeader: {
    color: "#002f5e",
  },
  details: {
    "&:nth-child(odd)": {
      backgroundColor: "#F4F6F8",
    },
    "@media (max-width: 420px)": {
      display: "block !important",
    },
  },
  listItem: {
    "@media (max-width: 420px)": {
      marginBottom: "10px",
    },
  },
  list: {
    width: "100%",
    padding: "0px !important",
  },
  availabilityCard: {
    paddingTop: "5px",
    textAlign: "center",
  },
  cartBtnRow: {
    padding: "16px",
  },
  unavailableText: {
    padding: "16px",
    textAlign: "justify",
  },
  moreStrainsCard: {
    marginTop: "16px",
    padding: "5px",
    textAlign: "center",
  },
  options: {
    paddingBottom: "8px",
    display: "flex",
    justifyContent: "center",
  },
  secondaryText: {
    paddingTop: "8px",
  },
  link: {
    textDecoration: "none",
    color: palette.primary.main,
  },
  checkoutBtn: {
    borderStyle: "none !important",
    borderRadius: "0px !important",
    fontSize: "0.9rem !important",
  },
  arrowIcon: {
    marginRight: "8px",
  },
  id: {
    display: "flex",
    justifyContent: "center",
  },
  snackbar: {
    width: "300px",
    backgroundColor: green[600],
    fontSize: "0.8rem",
    display: "flex",
    alignItems: "center",
    paddingTop: "0px",
    paddingBottom: "0px",
  },
  checkIcon: {
    fontSize: "0.8rem",
    marginRight: spacing(1),
  },
  closeIcon: {
    fontSize: "0.9rem",
  },
  unavailableIcon: {
    marginRight: spacing(2),
  },
  copyIcon: {
    fontSize: "0.8rem",
    "&:hover": {
      backgroundColor: "transparent",
      color: "rgba(0, 0, 0, 0.87)",
    },
  },
  detailsItem: {
    fontWeight: 420,
  },
  relatedItem: {
    display: "flex",
    justifyContent: "center !important",
  },
  addToCartBtn: {
    backgroundColor: green[600],
    color: "#fff",
    paddingLeft: "16px",
    paddingRight: "16px",
    "&:hover": {
      backgroundColor: green[700],
      color: "#fff",
    },
  },
  removeBtn: {
    marginTop: "5px",
  },
  quantity: {
    display: "flex",
    justifyContent: "space-evenly",
    alignItems: "center",
    padding: "16px",
  },
  maxItems: {
    color: palette.secondary.main,
  },
}))

export default useStyles
