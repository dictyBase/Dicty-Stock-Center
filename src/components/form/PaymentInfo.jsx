import React, { Component, PropTypes } from 'react'
import { reduxForm } from 'redux-form'
import Consumer from './Consumer'
import User from './User'
import 'styles/core.scss'

export const fields = [ 'firstName', 'lastName', 'email', 'org', 'group',
    'address', 'address2', 'city', 'state', 'zip', 'country', 'phone',
    'shipAccount', 'shipAccountNum', 'comments' ]

class PaymentInfo extends Component {
    displayName = 'payment information'

    static propTypes = {
        order: PropTypes.object
    }

    render() {
        const { consumer } = this.props.order
        const { editShipping } = this.props.orderActions
        const {
            fields: { firstName, lastName, email, org, group, address, address2, city,
                state, zip, country, phone }
        } = this.props
        return (
            <div>
                <div className="row">
                    <div className="col-xs-12">
                        <h2 className="page-header">Please enter payment information</h2>
                    </div>
                </div>
                <div className="row">
                    <div className="col-xs-12">
                        <Consumer consumer={ consumer } edit={ editShipping } />
                    </div>
                </div>
                <hr />
                <div className="row">
                    <form onSubmit={ '' } className="form-horizontal">
                        <div className="col-md-6">
                            <User title = { 'Payer Address' }
                                firstName = { firstName }
                                lastName = { lastName }
                                email = { email }
                                org ={ org }
                                group = { group }
                                address = { address }
                                address2 = { address2 }
                                city = { city }
                                state = { state }
                                zip = { zip }
                                country = { country }
                                phone = { phone }
                            />
                        </div>
                        <div className="col-md-6">
                            Payment Method
                        </div>
                    </form>
                </div>
            </div>
        )
    }
}

export default reduxForm({
    form: 'payment',
    fields
})(PaymentInfo)

