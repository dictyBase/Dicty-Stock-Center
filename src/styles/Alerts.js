import styled from "styled-components"

export const AlertBox = styled.div`
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #a94442;
  background-color: #f2dede;
  border-color: #ebccd1;
  text-align: center;
`

export const AlertSuccess = styled.div`
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #3c763d;
  background-color: #dff0d8;
  border-color: #d6e9c6;
  text-align: center;
`

export const AlertInfo = styled.div`
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #31708f;
  background-color: #d9edf7;
  border-color: #bce8f1;
`
