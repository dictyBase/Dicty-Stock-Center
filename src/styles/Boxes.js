import styled from "styled-components"
import { Box } from "rebass"

export const BorderBox = styled(Box)`
  border: 1px solid grey;
`

export const CenteredBox = styled(Box)`
  text-align: center;
`
