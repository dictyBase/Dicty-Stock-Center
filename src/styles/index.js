import styled from 'styled-components'

export const Container = styled.div`
    padding-right: 15px;
    padding-left: 15px;
    margin-right: auto;
    margin-left: auto;
    @media (min-width: 768px) {
        .container {
          width: 750px;
        }
      }
      @media (min-width: 992px) {
        .container {
          width: 970px;
        }
      }
      @media (min-width: 1200px) {
        .container {
          width: 1170px;
        }
      }
`
export const Grid = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    min-width: 100%;
`

// export const Row = styled.div`
//     &::after {
//       content: '';
//       clear: both;
//       display: table;
//     }
// `

// function getWidthString(span) {
//     if (!span) return

//     let width = span / 12 * 100
//     return `width: ${width}%`
// }

// export const Column = styled.div`
//     float: left;
//     ${({ xs }) => (xs ? getWidthString(xs) : 'width: 100%')}

//   @media only screen and (min-width: 768px) {
//     ${({ sm }) => sm && getWidthString(sm)}
//   }

//   @media only screen and (min-width: 992px) {
//     ${({ md }) => md && getWidthString(md)}
//   }

//   @media only screen and (min-width: 1200px) {
//     ${({ lg }) => lg && getWidthString(lg)}
//   }
// `

// export const Row = styled.div`
//     display: inline-block;
//     width: 100%;

//     /* @media (min-width: ${style.breakpoint.small}) {
//       display: flex;
//     } */
// `

// export const Column = styled.div`
//     display: inline-block;
//     /* margin-bottom: ${style.margin} */
//     margin-right: 0;
//     width: 100%;

//     /* @media (min-width: ${style.breakpoint.small}) {
//       margin-right: ${(props) => props.last ? '0' : style.margin};
//       flex: ${(props) => props.colspan}
//     } */
// `
