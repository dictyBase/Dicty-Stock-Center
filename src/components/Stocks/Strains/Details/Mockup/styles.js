import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles({
  root: {
    width: "80%",
    margin: "auto",
  },
  header: {
    textAlign: "center",
  },
  leftCard: {},
  details: {
    "&:hover": {
      boxShadow:
        "inset 1px 0 0 #dadce0,inset -1px 0 0 #dadce0,0 1px 2px 0 rgba(60,64,67,.3),0 1px 3px 1px rgba(60,64,67,.15)",
      zIndex: 1,
    },
    "&:nth-child(odd)": {
      backgroundColor: "#eeeeee",
    },
  },
  listTitle: {},
  listContent: {},
  rightCard: {
    padding: "5px",
    textAlign: "center",
  },
  quantity: {
    display: "inline-flex",
    justifyContent: "center",
  },
  textField: {},
})

export default useStyles
