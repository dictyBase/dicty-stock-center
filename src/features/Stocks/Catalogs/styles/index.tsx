import { makeStyles, Theme } from "@material-ui/core/styles"
import { blue } from "@material-ui/core/colors"

const useStyles = makeStyles((theme: Theme) => ({
  appBar: {
    backgroundColor: blue[800],
  },
  catalogPaper: {
    height: 600,
    width: "100%",
  },
  listHeaders: {
    position: "sticky",
    top: 0,
  },
  row: {
    borderBottom: "1px solid rgba(224, 224, 224, 1)",
    "&:hover": {
      backgroundColor: "#eeeeee",
      boxShadow:
        "inset 1px 0 0 #dadce0,inset -1px 0 0 #dadce0,0 1px 2px 0 rgba(60,64,67,.3),0 1px 3px 1px rgba(60,64,67,.15)",
      zIndex: 1,
    },
    "@media (max-width: 1024px)": {
      "& p": {
        fontSize: "0.75rem !important",
      },
    },
  },
  item: {
    paddingRight: theme.spacing(1),
  },
  trashBtn: {
    color: theme.palette.error.main,
  },
}))

export default useStyles
