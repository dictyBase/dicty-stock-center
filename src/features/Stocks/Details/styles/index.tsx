import { makeStyles, Theme } from "@material-ui/core/styles"

// @ts-ignore <-- necessary to include the !important attributes
const useStyles = makeStyles(({ palette, spacing }: Theme) => ({
  header: {
    textAlign: "center",
  },
  cardHeader: {
    color: "#002f5e",
    paddingRight: "8px",
  },
  details: {
    "&:nth-child(odd)": {
      backgroundColor: "#F4F6F8",
    },
    "@media (max-width: 420px)": {
      display: "block !important",
    },
  },
  listItem: {
    "@media (max-width: 420px)": {
      marginBottom: "10px",
    },
  },
  list: {
    width: "100%",
    padding: "0px !important",
  },
  id: {
    display: "flex",
    justifyContent: "center",
  },
  snackbar: {
    width: "300px",
    backgroundColor: palette.secondary.main,
    fontSize: "0.8rem",
    display: "flex",
    alignItems: "center",
    paddingTop: "0px",
    paddingBottom: "0px",
  },
  checkIcon: {
    fontSize: "0.8rem",
    marginRight: spacing(1),
  },
  closeIcon: {
    fontSize: "0.9rem",
  },
  copyIcon: {
    fontSize: "0.8rem",
    "&:hover": {
      backgroundColor: "transparent",
      color: "rgba(0, 0, 0, 0.87)",
    },
  },
  detailsItem: {
    fontWeight: 420,
  },
  maxItems: {
    color: palette.error.main,
  },
  moreInfoText: {
    marginTop: "4px",
    fontSize: "0.75rem",
    color: "#000",
  },
  error: {
    color: palette.error.main,
  },
}))

export default useStyles
