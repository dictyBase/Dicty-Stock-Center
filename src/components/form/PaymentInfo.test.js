import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import PaymentInfo from "./PaymentInfo"
import { shallow } from "enzyme"
import { FormControl, HelpBlock } from "styles"

describe("form/PaymentInfo", () => {
  const payMethod = { value: "Credit", touched: true }
  const poNum = { value: "", touched: true }

  const wrapper = shallow(<PaymentInfo payMethod={payMethod} poNum={poNum} />)

  it("should render poNum textbox only when payMethod is PO", () => {
    expect(wrapper.containsMatchingElement(<FormControl />)).toEqual(false)

    wrapper.setProps({ payMethod: { value: "PO", touched: true } })

    expect(wrapper.containsMatchingElement(<FormControl />)).toEqual(true)
  })

  it("should render validation error messages properly", () => {
    const field = { value: "", touched: true, error: "required" }
    wrapper.setProps({ payMethod: field })
    expect(wrapper.contains(<HelpBlock>{field.error}</HelpBlock>)).toEqual(true)
  })
})

test("matching a snapshot of PaymentInfo", () => {
  const payMethod = {
    value: "Credit",
    touched: true
  }
  const poNum = {
    value: "",
    touched: true
  }

  const component = renderer.create(
    <PaymentInfo payMethod={payMethod} poNum={poNum} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
