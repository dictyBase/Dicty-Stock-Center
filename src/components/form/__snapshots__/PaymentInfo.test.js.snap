// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of PaymentInfo 1`] = `
<div
  className="sc-kLIISr itrVsn"
>
  <div
    className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa cyncUw"
      is={null}
    >
      <label
        className="sc-dliRfk eLcvEu"
      >
        <span
          className="sc-hmXxxW fwNEuV"
          title="required field"
        >
          * 
        </span>
        Payment Method:
      </label>
    </div>
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa XupaC"
      is={null}
    >
      <label
        className="sc-bmyXtO ddZAYX"
      >
        <input
          checked={false}
          touched={true}
          type="radio"
          value="Credit card"
        />
          Credit Card
      </label>
      <label
        className="sc-bmyXtO ddZAYX"
      >
        <input
          checked={false}
          touched={true}
          type="radio"
          value="Wire transfer"
        />
          Wire Transfer
      </label>
      <label
        className="sc-bmyXtO ddZAYX"
      >
        <input
          checked={false}
          touched={true}
          type="radio"
          value="PO"
        />
          Purchase Order (PO)
      </label>
    </div>
  </div>
</div>
`;
